xds/internal/xdsclient/tests/cds_watchers_test.go

```diff
@@ -645,12 +702,30 @@ func (s) TestCDSWatch_ValidResponseCancelsExpiryTimerBehavior(t *testing.T) {
 //     callback associated with that resource.  It should not result in the
 //     invocation of the watch callback associated with the deleted resource.
 func (s) TestCDSWatch_ResourceRemoved(t *testing.T) {
-	t.Skip("Disabled; see https://github.com/grpc/grpc-go/issues/6781")
-	mgmtServer, nodeID, bootstrapContents, _, cleanup := e2e.SetupManagementServer(t, e2e.ManagementServerOptions{})
-	defer cleanup()
+	mgmtServer := e2e.StartManagementServer(t, e2e.ManagementServerOptions{})
+
+	nodeID := uuid.New().String()
+	bc, err := bootstrap.NewContentsForTesting(bootstrap.ConfigOptionsForTesting{
+		Servers: []json.RawMessage{[]byte(fmt.Sprintf(`{
+					"server_uri": %q,
+					"channel_creds": [{"type": "insecure"}]
+				}`, mgmtServer.Address))},
+		NodeID: nodeID,
+		Authorities: map[string]json.RawMessage{
+			// Xdstp style resource names used in this test use a url escaped
+			// version of t.Name as their authority, and the empty config
+			// results in the top-level xds server configuration being used for
+			// this authority.
+			url.PathEscape(t.Name()): []byte(`{}`),
```

Comment:
I'm thinking if PathEscape is the right transformer for `t.Name` because, we have path escapes of different pieces of the dialTarget. I'm concerned if this has unknown sideeffects? 

How about maybe just trimming all the `/`s from t.Name() instead?

Reply:
Done.